### PR TITLE
[94] Browse auctions (Frontend)

### PR DESCRIPTION
**Pull request by SaidganiUsm**
Where
[Browse auctions (Frontend)](https://jjbcapital.atlassian.net/browse/NL-94)

What
See all lost in the auction

How
Browsing the auction

Affected Area
Auctionify.API

PR Checklist
Build

 Build Succeeded
Unit Tests

 X Tests Passed, Y Ignored, Z Failed

### CODE REVIEW

**Comment by khasanrashidov** [comment]
some things are to be fixed:

1) icons are not showing properly

image

2) browse auctions (get all lots), horizontally need to be displayed, and wishlist icon should be displayed properly as well:

image

**Comment by khasanrashidov** [comment]
these things are to be fixed:

image


image

**Comment by khasanrashidov** [comment]
Please, Saidgani, icons are to be corrected

image


image


image

File: src/Auctionify.API/ClientApp/src/app/components/buyer/buyer-routing.module.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { RouterModule, Routes } from '@angular/router';
+import { NgModule } from '@angular/core';
+import { FilterComponent } from './filter/filter.component';
+
+const sellerRoutes: Routes = [
```
Comment (khasanrashidov) [inline src/Auctionify.API/ClientApp/src/app/components/buyer/buyer-routing.module.ts:5]: I think it is buyerRoutes

**Review by otash8110 — APPROVED** [review]
Minor differences with design. Overall implementation seems fine for me

**Review by TsvirkoNikita — CHANGES_REQUESTED** [review]
image
- remove 'Example' links
- sorting does not work
- pagination button for active lots does not work
- all auctions are always pulled from the backend, despite the filter endpoint support pagination.
- if a row of lots does not fit on the screen, the lots should be moved to the next row. There should be no horizontal scrolling on the page

**Review by otash8110 — CHANGES_REQUESTED** [review]
image
- Remove example links on top
- Lots still do not wrap, it creates horizontal scroll on screen sizes 1400px. I Suggest using flexbox
- Sort button still not sorting but just has a pointer cursor. I suggest removing it if there is no this functionality

image
- I think it is a good idea to make those buttons invisible when there are no lots in any of the given statuses. 

image
- Fix text style for lots and make other lot statuses' lots also clickable as Active lots

**Comment by khasanrashidov** [comment]
1. When I configured the price only in filter (set the range max)... the upcoming lot also was shown to me.. but I didn't check the upcoming lot checkbox in filter:
filter:
image

what I got (that one within red is upcoming lot):
image

2. Bid count didn't load, although in DB for lots, I have them

image


3. Can you add hovering effect with shadow +  cursor: pointer, because here we don't have borders in design, so for user it would be convenient to browse lots:

image

4. Can you add watchlist (wishlist) functionality? So buyer (who logged in), can add them. The watchlist functionality is currently available in dev_111

image_2023-12-08_20-44-27

**Comment by khasanrashidov** [comment]
### 1. Resolve conflicts.

### 2. Remove that, we are returning currency code from API (what if you have roubles, or soums in currency?)
image

### 3. I would a bit space between them:
image

**Review by otash8110 — APPROVED** [review]

**Review by TsvirkoNikita — CHANGES_REQUESTED** [review]
- For each auction, if it has bids, you need to show its actual price, not the starting price
- Why did you remove the sort button?
- The filter is applied only to the upper section with active lots and in this case lots in different statuses start to be displayed there. The filter should be applied to each section. If we select, for example, only active lots in the filter, then only the upper section with active lots will be displayed
- Lots row still not responsive:
image

**Comment by otash8110** [comment]
> * For each auction, if it has bids, you need to show its actual price, not the starting price
> * Why did you remove the sort button?
> * The filter is applied only to the upper section with active lots and in this case lots in different statuses start to be displayed there. The filter should be applied to each section. If we select, for example, only active lots in the filter, then only the upper section with active lots will be displayed
> * Lots row still not responsive:
>   image

Fixed filter

**Review by TsvirkoNikita — CHANGES_REQUESTED** [review]
- Price filtering still using the starting price of the lot, instead the current price
- Selecting a specific status in the filter has no effect
image

**Review by TsvirkoNikita — APPROVED** [review]

**Review by otash8110 — APPROVED** [review]

**Pull request merged**